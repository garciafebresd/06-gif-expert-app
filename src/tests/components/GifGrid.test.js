import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Testing GifGrid component', () => {

    const category = 'saitama gif';

    test('should mach with snapshot', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });
    
    test('should show items when the images loaded useFetchGifs', () => {
      
        const gifs = [{
            id: 'DFDF',
            url: 'https://localhost/image_test.jpg',
            title: 'Cualquier cosa'
        },
        {
            id: '123',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category} />);

        // expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
    

});


