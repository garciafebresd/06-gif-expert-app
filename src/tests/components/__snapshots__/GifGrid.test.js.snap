// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGrid component should mach with snapshot 1`] = `
<Fragment>
  <h3
    key="saitama gif"
  >
    saitama gif
  </h3>
  <p
    className="card animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid card animate__animated animate__fadeIn"
  />
</Fragment>
`;
